pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -1284,8 +1284,27 @@ public void addSchema(Schema schema, boolean override, boolean force)
     }
   }
 
+  public void updateSchemaDateTime(TableConfig tableConfig, Schema schema)
+      throws SchemaNotFoundException, SchemaBackwardIncompatibleException, TableNotFoundException {
```

Comment:
It shouldn't throw these exceptions, or simply pass `tableNameWithType` and throws these exceptions. Caller should check the exception type to create the `ControllerApplicationException` accordingly